### PR TITLE
Added docs for cache: no-store

### PR DESCRIPTION
**Pull request by tewaro**
### Summary

Update Documentation for adding `cache: no-store`.

### Documentation checklist

- [ ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [ ] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.
- [ ] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

File: src/content/compatibility-dates/cache-no-store.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+When this flag is enabled, the cache header becomes available within the [`fetch()` API](/workers/runtime-apis/fetch/). The option for `no-store` is enabled.
+
+For a cross-zone request this simply specifies the headers `Pragma: no-cache` and `Cache-Control: no-cache` to the origin.
+
+For grey-cloud subrequests this bypasses the Cloudflare's cache.
+
+Example:
+
+```js
+const response = await fetch("https://example.com", { cache: 'no-store'});
```
Comment (mhart) [inline src/content/compatibility-dates/cache-no-store.md:31]: Suggestion: add an example for `new Request()` too? We support it there too right?

File: src/content/compatibility-dates/cache-no-store.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	list: never
+
+name: "Enable `cache: no-store` HTTP standard API"
+sort_date: "2024-11-11"
+enabled_date: "2024-11-11"
+enable_flag: "cache_option_enabled"
+disable_flag: "cache_option_disabled"
+---
+
+When this flag is enabled, the cache header becomes available within the [`fetch()` API](/workers/runtime-apis/fetch/). The option for `no-store` is enabled.
```
Comment (irvinebroque) [inline src/content/compatibility-dates/cache-no-store.md]: What this enables is this:

https://developer.mozilla.org/en-US/docs/Web/API/Request/cache

The `cache` property of `Request`.

I see where you're going describing it in terms of headers, but "the header becomes available" not quite right.

We should be precise here and elsewhere:

```suggestion
When you enable the `cache_option_enabled` compatibility flag, you can specify a value for the `cache` property of the `Request` interface. When this compatibility flag is not enabled, or `cache_option_disabled` is set, the Workers runtime will throw an error if you attempt to specify the `cache` property.

For example, when this flag is enabled you can instruct Cloudflare not to cache the response from a subrequest you make from your Worker, using the [`fetch()` API](/workers/runtime-apis/fetch/):

<example using no-store>

...then show how this maps to headers, as you do below on line 16
```

File: src/content/docs/workers/examples/cache-using-fetch.mdx [hunk 3]
```diff
@@ -466,3 +466,15 @@ async function handleRequest(request) {
 ```
 
 </TabItem> </Tabs>
+
+## Using the HTTP Cache API
+
+Currently we only support the `no-store` header for controlling cache.
```
Comment (irvinebroque) [inline src/content/docs/workers/examples/cache-using-fetch.mdx]: This is out of context, need to back up here

File: src/content/docs/workers/examples/cache-using-fetch.mdx [hunk 4]
```diff
@@ -466,3 +466,15 @@ async function handleRequest(request) {
...
+When `no-store` is supplied the cache is bypassed both on the way to the origin and on the way back.
+For cross-zone requests this simply forwards the `Pragma: no-cache` and `Cache-Control: no-cache` headers to the origin.
+This is because the cache is on the origin side for cross-zone requests.
+For grey-cloud requests this bypasses the use of Cloudflare's caches.
+
+```js
+fetch(event.request, { cache: 'no-store'});
```
Comment (irvinebroque) [inline src/content/docs/workers/examples/cache-using-fetch.mdx]: Why does this use `event.request`? That seems like leftover service worker thing?

File: src/content/docs/workers/examples/cache-using-fetch.mdx [hunk 5]
```diff
@@ -466,3 +466,15 @@ async function handleRequest(request) {
 ```
 
 </TabItem> </Tabs>
+
+## Using the HTTP Cache API
+
+Currently we only support the `no-store` header for controlling cache.
+When `no-store` is supplied the cache is bypassed both on the way to the origin and on the way back.
+For cross-zone requests this simply forwards the `Pragma: no-cache` and `Cache-Control: no-cache` headers to the origin.
```
Comment (irvinebroque) [inline src/content/docs/workers/examples/cache-using-fetch.mdx]: This example content shouldn't be our reference material here.

This kind of content should be here:

https://developers.cloudflare.com/workers/runtime-apis/fetch/

File: src/content/docs/workers/runtime-apis/request.mdx [hunk 6]
```diff
@@ -62,7 +62,9 @@ let request = new Request(input, options)
 
 An object containing properties that you want to apply to the request.
 
+* `cache` `undefined | 'no-store'` optional
 
+	* Standard HTTP `cache` header. Only `cache: no-store` is supported.
```
Comment (irvinebroque) [inline src/content/docs/workers/runtime-apis/request.mdx]: Can we explain what the behavior here is if you attempt to set a value other than `no-store`?

**Review by irvinebroque — COMMENTED** [review]
@mikenomitch FYI

File: src/content/compatibility-dates/cache-no-store.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+For a cross-zone request this simply specifies the headers `Pragma: no-cache` and `Cache-Control: no-cache` to the origin.
+
+For grey-cloud subrequests this bypasses the Cloudflare's cache.
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md]: We sometimes use grey cloud in external docs, but for the most part I think we shouldn't. We know what it means internally, but average new user won't.

File: src/content/compatibility-dates/cache-no-store.md (hunk 2 above)
Comment (mikenomitch) [reply src/content/compatibility-dates/cache-no-store.md]: +1 to the wording change. Was a bit hard to follow exactly what you can now do originally.

File: src/content/compatibility-dates/cache-no-store.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+name: "Enable `cache: no-store` HTTP standard API"
+sort_date: "2024-11-11"
+enabled_date: "2024-11-11"
+enable_flag: "cache_option_enabled"
+disable_flag: "cache_option_disabled"
+---
+
+When this flag is enabled, the cache header becomes available within the [`fetch()` API](/workers/runtime-apis/fetch/). The option for `no-store` is enabled.
+
+For a cross-zone request this simply specifies the headers `Pragma: no-cache` and `Cache-Control: no-cache` to the origin.
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md]: style nitpick: I'd replace "simply" with "just" or just drop the word.

Comment (mikenomitch) [reply src/content/compatibility-dates/cache-no-store.md]: Also, I think this reads slightly off "specifies headers to the origin" - specifies headers in the request to the origin?

File: src/content/docs/workers/examples/cache-using-fetch.mdx (hunk 3 above)
Comment (mikenomitch) [reply src/content/docs/workers/examples/cache-using-fetch.mdx]: +1

Maybe you start by explaining how the cache normally works (Brendan's next comment), and then explain the header?

File: src/content/docs/workers/examples/cache-using-fetch.mdx [hunk 9]
```diff
@@ -466,3 +466,15 @@ async function handleRequest(request) {
...
+This is because the cache is on the origin side for cross-zone requests.
+For grey-cloud requests this bypasses the use of Cloudflare's caches.
```
Comment (mikenomitch) [inline src/content/docs/workers/examples/cache-using-fetch.mdx]: Noting the other "grey-cloud" use (see above)

File: src/content/compatibility-dates/cache-no-store.md (hunk 7 above)
Comment (tewaro) [reply src/content/compatibility-dates/cache-no-store.md]: do we want to do the same with cross-zone?

File: src/content/docs/workers/examples/cache-using-fetch.mdx (hunk 9 above)
Comment (tewaro) [reply src/content/docs/workers/examples/cache-using-fetch.mdx]: Same question here about cross-zone.

File: src/content/compatibility-dates/cache-no-store.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+name: "Enable `cache: no-store` HTTP standard API"
+sort_date: "2024-11-11"
+enabled_date: "2024-11-11"
+enable_flag: "cache_option_enabled"
+disable_flag: "cache_option_disabled"
+---
+
+When you enable the `cache_option_enabled` compatibility flag, you can specify a value for the `cache` property of the Request interface.
+When this compatibility flag is not enabled, or `cache_option_disabled` is set, the Workers runtime will throw an `Error` saying `The 'cache' field on
+'RequestInitializerDict is not implemented.`
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md]: nit - I think missing a ' after RequestInitializerDict (or added one before it inadvertantly

File: src/content/compatibility-dates/cache-no-store.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+Examples using `cache: 'no-store'`.
+
+```js
+const response = await fetch("https://example.com", { cache: 'no-store'});
+```
+
+You can also create a Response object to pass around:
+
+```js
+const request = new Response("https://example.com", { cache: 'no-store'});
```
Comment (mhart) [inline src/content/compatibility-dates/cache-no-store.md]: This should be `Request`, not `Response`

File: src/content/compatibility-dates/cache-no-store.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+enabled_date: "2024-11-11"
+enable_flag: "cache_option_enabled"
+disable_flag: "cache_option_disabled"
+---
+
+When you enable the `cache_option_enabled` compatibility flag, you can specify a value for the `cache` property of the Request interface.
+When this compatibility flag is not enabled, or `cache_option_disabled` is set, the Workers runtime will throw an `Error` saying `The 'cache' field on
+'RequestInitializerDict is not implemented.`
+
+For example, when this flag is enabled you can instruct Cloudflare not to cache the response from a subrequest you make from your Worker, using the [`fetch()` API](/workers/runtime-apis/fetch/):
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md]: The "For example" feels a bit odd here because that's more or less the only thing you can do at this point right. Maybe just drop that?

And then add comma after "when this flag is enabled"

Also, I think the comma before "using" can be dropped?

File: src/content/compatibility-dates/cache-no-store.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+
+The only cache option enabled with `cache_option_enabled` is `'no-store'`.
+Any other value will cause the Workers runtime to throw a `TypeError` with the message `Unsupported cache mode: <the-mode-you-specified>`.
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md]: nit - maybe: "Specifying any other value will throw a throw a `TypeError` with the message `Unsupported cache mode: <the-mode-you-specified>`."

File: src/content/compatibility-dates/cache-no-store.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+When `no-store` is specified:
+* For a cross-zone request the headers `Pragma: no-cache` and `Cache-Control: no-cache` are sent to the origin.
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md]: I try to make the bullet points read well with the header

If you read it like "When no-store is specified for a cross zone", the "for" sounds kind of weird

So I'd do:

When `no-store` is specified:
* All requests have the headers `Pragma: no-cache` and `Cache-Control: no-cache` set
* Subrequests to origins outside of Cloudflare bypass Cloudflare's cache

File: src/content/compatibility-dates/cache-no-store.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+* For subrequests to origins not hosted by Cloudflare, `no-store` bypasses Cloudflare's cache in addition to specifying the above headers.
+
+Examples using `cache: 'no-store'`.
+
+```js
+const response = await fetch("https://example.com", { cache: 'no-store'});
+```
+
+You can also create a Response object to pass around:
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md]: Maybe just: "The cache value can also be set on Request"

"to pass around" isn't doing much here

File: src/content/compatibility-dates/cache-no-store.md [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+Any other value will cause the Workers runtime to throw a `TypeError` with the message `Unsupported cache mode: <the-mode-you-specified>`.
+
+When `no-store` is specified:
+* For a cross-zone request the headers `Pragma: no-cache` and `Cache-Control: no-cache` are sent to the origin.
+
+* For subrequests to origins not hosted by Cloudflare, `no-store` bypasses Cloudflare's cache in addition to specifying the above headers.
+
+Examples using `cache: 'no-store'`.
+
+```js
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md:30]: Maybe add a little text about this one specifically, so it's similar to to the one below.

Maybe just: "The 'cache' have can be passed into the options for `fetch`"

File: src/content/compatibility-dates/cache-no-store.md [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+
+The only cache option enabled with `cache_option_enabled` is `'no-store'`.
+Any other value will cause the Workers runtime to throw a `TypeError` with the message `Unsupported cache mode: <the-mode-you-specified>`.
+
+When `no-store` is specified:
+* For a cross-zone request the headers `Pragma: no-cache` and `Cache-Control: no-cache` are sent to the origin.
+
+* For subrequests to origins not hosted by Cloudflare, `no-store` bypasses Cloudflare's cache in addition to specifying the above headers.
+
+Examples using `cache: 'no-store'`.
```
Comment (mikenomitch) [inline src/content/compatibility-dates/cache-no-store.md]: This isn't a sentence, so we shouldn't have a period. Maybe our lowest lever of header would render well? Or just replace the period with a ":"?

File: src/content/docs/workers/examples/cache-using-fetch.mdx [hunk 18]
```diff
@@ -466,3 +466,13 @@ async function handleRequest(request) {
 ```
 
 </TabItem> </Tabs>
+
+## Using the HTTP Cache API
+
+The `cache` mode can be set on fetches.
```
Comment (mikenomitch) [inline src/content/docs/workers/examples/cache-using-fetch.mdx]: "can be set in `fetch` options" maybe? - A little more specific

File: src/content/docs/workers/examples/cache-using-fetch.mdx [hunk 19]
```diff
@@ -466,3 +466,13 @@ async function handleRequest(request) {
...
+Currently we only support the `no-store` mode for controlling the cache.
```
Comment (mikenomitch) [inline src/content/docs/workers/examples/cache-using-fetch.mdx]: s/we only support/Workers only supports

File: src/content/docs/workers/examples/cache-using-fetch.mdx [hunk 20]
```diff
@@ -466,3 +466,13 @@ async function handleRequest(request) {
...
+When `no-store` is supplied the cache is bypassed both on the way to the origin and on the way back.
```
Comment (mikenomitch) [inline src/content/docs/workers/examples/cache-using-fetch.mdx]: What does "on the way back" mean in this case?

File: src/content/docs/workers/runtime-apis/fetch.mdx [hunk 21]
```diff
@@ -75,6 +75,12 @@ async function eventHandler(event) {
 * [`resource`](https://developer.mozilla.org/en-US/docs/Web/API/fetch#resource) Request | string | URL
 
 * `options` options
+	* `cache` `undefined | 'no-store'` optional
+		* Standard HTTP `cache` header. Only `cache: 'no-store'` is supported.
+		Any other `cache` header will result in a `TypeError` with the message `Unsupported cache mode: <attempted-cache-mode>`.
+			* For cross-zone requests this simply forwards the `Pragma: no-cache` and `Cache-Control: no-cache` headers to the origin.
+			This is because the cache is on the origin side for cross-zone requests.
+			* For requests to origins not hosted by Cloudflare, `no-store` bypasses the use of Cloudflare's caches.
```
Comment (mikenomitch) [inline src/content/docs/workers/runtime-apis/fetch.mdx:82]: Its a bit unclear reading this if non-Cloudflare get the headers or not. Reading above it sounds like it does.

**Review by mikenomitch — CHANGES_REQUESTED** [review]

File: src/content/docs/workers/runtime-apis/fetch.mdx (hunk 21 above)
Comment (tewaro) [reply src/content/docs/workers/runtime-apis/fetch.mdx:82]: They get the headers.

Comment (tewaro) [reply src/content/docs/workers/runtime-apis/fetch.mdx:82]: I will fix the wording.

**Review by mikenomitch — APPROVED** [review]
LGTM now - thanks for making the changes
